RNA-seq/06-openpbta_heatmap.Rmd

```diff
@@ -0,0 +1,430 @@
+# Find the value that we'll use as a threshold to filter the top 10%
+variance_threshold <- quantile(gene_variance, 0.9)
+# Row indices of high variance genes
+high_variance_index <- which(gene_variance > variance_threshold)
+# What does a row index look like?
+head(high_variance_index)
+```
+
+```{r high_var_mat, live = TRUE}
+# Get a matrix that is subset to just the high variance genes
+high_var_mat <- expr_mat[high_variance_index, ]
+```
+
+## Heatmap
+
+### Annotation
+
+First, we'll set up the metadata that we want to use to label samples in the heatmap.
+In `ComplexHeatmap` terminology, this is called annotation, or `HeatmapAnnotation`, specifically.
+
+```{r sample_annotation_df}
+sample_annotation_df <- histologies_df %>%
+  # Select only the columns that we'll use
+  select(Kids_First_Biospecimen_ID,
+         short_histology,
+         composition,
+         tumor_descriptor)
+
+# Let's examine these columns
+sample_annotation_df
+```
+`ComplexHeatmap` is going to want the data frame we provide to have the sample identifiers as row names, so let's set that up. 
+
+```{r annotation_as_df, live = TRUE}
+sample_annotation_df <- sample_annotation_df %>%
+  tibble::column_to_rownames("Kids_First_Biospecimen_ID")
+```
+
+To specify the colors in our annotation bar, we need to create a list of named vectors.
+The names of the list need to *exactly* match the column names in `sample_annotation_df` and the names in each vector need to *exactly* match the values in those columns.
+
+```{r color_palettes}
+# The Okabe Ito palette is recommended for those with color vision deficiencies  
+histology_colors <- palette.colors(palette = "Okabe-Ito")[2:5]
+# `palette.colors()` returns a named vector, which can cause trouble
+histology_colors <- unname(histology_colors)
+
+# annotation color list for ComplexHeatMap
+sample_annotation_colors <- list(
+  short_histology = c(
+    "LGAT" = histology_colors[[1]],
+    "Ependymoma" = histology_colors[[2]],
+    "HGAT" = histology_colors[[3]],
+    "Medulloblastoma" = histology_colors[[4]]
+  ),
+  composition = c(
+    "Solid Tissue" = "#A0A0A0",  # light grey
+    "Derived Cell Line" = "#000000"  # black
+  ),
+  tumor_descriptor = c(
+    "Initial CNS Tumor" = "#3333FF",
+    "Progressive" = "#FFFF99",
+    "Recurrence" = "#CCCCFF",
+    "Second Malignancy" = "#000033",
+    "Unavailable" = "#FFFFFF"  # white for missing data
+  )
+)
+```
+
+We need to create a special type of object with `HeatmapAnnotation` using the annotation data frame and the list of color vectors.
+
+```{r create_annotation, live = TRUE}
+column_annotation <- HeatmapAnnotation(
+  df = sample_annotation_df,
+  col = sample_annotation_colors
+)
+```
+
+### Values for display
+
+We will z-score the expression values for display.
+This is sometimes called a standardized score.
+It's calculated for each value in a row (gene) by subtracting the gene's mean and dividing by the gene's standard deviation.
+
+```{r}
+zscores_mat <- 
```

Comment:
I guess before I say that I should ask: What does the heatmap look like without scaling?

Reply:
Eh, it isn't great: 
![image](https://user-images.githubusercontent.com/20483/110190958-08ea5800-7df4-11eb-91d7-dd415f9ed048.png)

I bet I can tighten up the z score explanation though.